docs/integrations/web.md

```diff
@@ -36,209 +36,177 @@ There are a couple important concepts to grasp for using Armory. We’ll look at
+    getConnection("armory"),
+    new Screen("HelloScreen").add(new Title("Hello there!"))
+);
+```
 
-The `GenericTemplate` is quite packed, but it can be much simpler than that - for example, we also provide a `TitleTemplate` which literally only contains a TitleComponent.
+But let's first learn how to create those screens!
 
-The components are the building blocks of screens, and there are several of them predefined in the service, all implementing the `BaseComponent` interface:
-* ActionGroup (groups together buttons into a list of options out of which only one can be selected)
-* BackButton
-* CloudSelect (allows user to select multiple elements from a list)
+### Components
+The components are the building blocks of screens, and there are several you can use out-of-the-box, all implementing the `BaseComponent` interface:
+* BackButton: component used in Screen headers
+* CloudSelect: a cool multi-select component
 * Description
+* Header: component that normally contains a logo and the `BackButton` (enabled by setting the `allowsBack` field to `true`)
 * Image
-* Input (roughly corresponds to HTML input element, with the data type specified by setting `type`)
-* SubmitButton (basic button, extending the `BaseSubmitButtonComponent` which triggers a `SubmitEvent`)
-* TextArea
+* Input: component that roughly corresponds to the HTML input element, with the input data type specified by setting `type`
+* SubmitButton: basic button component, triggers a `Submit` event
+* TextArea: component for longer text input
 * Title
 
-Each component is referenced through its `componentId`. We’ll use this id later on for getting the data the user provided on a screen off the `SubmitEvent`.
-
-## Template generator
-
-We mentioned you can always use one of the predefined templates to create screens, such as the `TitleTemplate`:
-```BaseTemplate screen = new TitleTemplate("Hello, world!");```
-
-But assuming you’ll often want to create your own layouts, we’ll now focus a bit more on the `TemplateGenerator`. Let’s look at an example of how we can use it to create a new template:
-
+Components that are used to collect some sort of input or activity from the user (text areas, buttons etc.) are referenced through the `inputId`. For example, here is a rule that registers the user entered their name and submitted it by pressing a button:
 ```java
-new TemplateGenerator("exampleTemplate")
-            .addComponent("title", new TitleComponent("Screen Title"))
-            .addComponent("description", new DescriptionComponent("Here is where we put the description."))
-            .addComponent("input", new InputComponent("name", "Type your name…", true))
-            .addComponent("actionGroup", new ActionGroupComponent(List.of(
-                        new PrimarySubmitButtonComponent("inputId1", "Option 1", "NextScreen1"),
-                        new PrimarySubmitButtonComponent("inputId2", "Option 2", "NextScreen2")
-)));
+rule "Save customer name"
```

Comment:
In the code above this, I would create a screen that has a title, input with id `name` and a submit button with id `submitName`. That way you can reference that here